packages/core/src/WebSocket.ts

```diff
@@ -1,41 +1,90 @@
+// cSpell:ignore utxos
+
+import { BlockNo } from './Cardano/types/Block';
 import { HealthCheckResponse, NetworkInfoProvider, Provider } from './Provider';
+import { HexBlob } from '@cardano-sdk/util';
+import { HydratedTx } from './Cardano/types/Transaction';
 import { Observable, ReplaySubject, firstValueFrom } from 'rxjs';
+import { PaymentAddress } from './Cardano/Address/PaymentAddress';
 
 export type AsyncReturnType<F extends () => unknown> = F extends () => Promise<infer R> ? R : never;
 
+export type MetadataDbModel = [string, HexBlob];
+
 export type NetworkInfoMethods = Exclude<keyof NetworkInfoProvider, 'healthCheck'>;
 export type NetworkInfoResponses = { [m in NetworkInfoMethods]: AsyncReturnType<NetworkInfoProvider[m]> };
 
 export interface WSMessage {
+  /** The addresses the client subscribes. */
+  txsByAddresses?: { addresses: PaymentAddress[]; lower: BlockNo };
+
   /** The client id assigned by the server. */
   clientId?: string;
 
+  /** The error on server while performing a request. */
+  error?: Error;
+
   /** Latest value(s) for the `NetworkInfoProvider` methods.*/
   networkInfo?: Partial<NetworkInfoResponses>;
+
+  /** The request id from client to server. */
+  requestId?: number;
+
+  /** This message is the response to the message with this id. */
+  responseTo?: number;
+
+  /** The server is still syncing. */
+  syncing?: boolean;
+
+  /** The transactions. */
+  transactions?: HydratedTx[];
+
+  /** The partial transactions for UTXOs. */
+  utxos?: HydratedTx[];
+}
+
+interface EmitHealthOptions {
+  notRecoverable?: boolean;
```

Comment:
Yes... in this case I used the negative form as it also has the meaning "this error needs to end up to a server restart".

Reply:
Sure there are exceptions to the rule :pray: 

On a different topic:
> has the meaning "this error needs to end up to a server restart".

In that case, server should probably die, and all ws connections get disconnected? :thinking:

Reply:
Yes... in case of not recoverable error there is no alternative...
The server tries to, and often it does... due to the nature of a not recoverable error, sometimes it could be something is not correctly closed and self shutdown is not successful; anyway the health check is set to inform the orchestrator it has to perform a forced shutdown.

Reply:
@gytis-ivaskevicius what is the preferred behavior for unrecoverable errors: die or report unhealthy?

Reply:
We discussed this in deep: we are working on health checks since some weeks ago.
The `ws-server` implements the health check logic as specified by SRE (two distinct endpoints one for readiness one for the health status).
We didn't specifically discuss the _self shutdown_ option, anyway if the required action is a restart, making the server able to do it by itself (rather than waiting for the orchestrator to do that) sounds as an improvement...
For sure, any insights from @gytis-ivaskevicius will be appreciated.

Reply:
if a restart may resolve the error - either liveliness probe should fail or it should exit with non 0 exit code. Service shutting down in this case is a better option since it gives a faster feedback loop

if the error can not be resolved by restarting - it would be better if liveliness probe would fail to reduce restarts and possibly keep other parts of application working